pkg/server/config.go

```diff
@@ -245,35 +251,34 @@ func GetTotalMemory() (int64, error) {
 				log.Infof(context.TODO(), "can't read available memory from cgroups (%s), using system memory %s instead", err,
 					humanizeutil.IBytes(totalMem))
 			}
-			return totalMem, nil
+			return checkTotal(totalMem)
 		}
 		var cgAvlMem uint64
 		if cgAvlMem, err = strconv.ParseUint(strings.TrimSpace(string(buf)), 10, 64); err != nil {
 			if log.V(1) {
 				log.Infof(context.TODO(), "can't parse available memory from cgroups (%s), using system memory %s instead", err,
 					humanizeutil.IBytes(totalMem))
 			}
-			return totalMem, nil
+			return checkTotal(totalMem)
 		}
 		if cgAvlMem > math.MaxInt64 {
 			if log.V(1) {
 				log.Infof(context.TODO(), "available memory from cgroups is too large and unsupported %s using system memory %s instead",
 					humanize.IBytes(cgAvlMem), humanizeutil.IBytes(totalMem))
 
 			}
-			return totalMem, nil
+			return checkTotal(totalMem)
 		}
-		if cgAvlMem > mem.Total {
+		if mem.Total > 0 && cgAvlMem > mem.Total {
```

Comment:
`mem.Total` is (intuitively, not by type) equal to `totalMem`, right? I think it would add to clarity if `mem` went out of scope after `totalMem` was a thing (even at the cost of a type conversion). Just a suggestion, might be misreading this as well.